### PR TITLE
feat(Core): Added a check job to ensure that our prod code is ES5 com…

### PR DESCRIPTION
**Pull request by akoushke**
This is my understanding  from the ticket Adam. Please let me know what you think of it.

### CODE REVIEW

**Comment by akoushke** [comment]
will def change the commit message.

File: Jenkinsfile [hunk 1]
```diff
@@ -186,6 +186,12 @@ ansiColor('xterm') {
             }
           }
 
+          stage('Check for ES5 Compatibility') {
+            sh'''
+              node_modules/.bin/es-check es5 packages/node_modules/@ciscospark/**/dist/*.js
```
Comment (adamweeks) [inline Jenkinsfile]: This should also check the `/esm` and `/cjs` folders. The `esm` folder is what webpack and other module loaders will load and aren't usually configured to transpile code. 

We should also add this step to the [`publish:components` helper](https://github.com/webex/react-ciscospark/blob/cd79c7d54bcd5728d9d68e5401d8af1fca67ae1c/Jenkinsfile#L246) to verify those esm and cjs folders are good

File: Jenkinsfile [hunk 2]
```diff
@@ -186,6 +186,12 @@ ansiColor('xterm') {
             }
           }
 
+          stage('Check for ES5 Compatibility') {
+            sh'''
```
Comment (adamweeks) [inline Jenkinsfile]: Could you add this to the end of the `Build for CDN` stage instead since it is checking the files built in that stage.

Comment (akoushke) [reply Jenkinsfile]: I thought i did.

File: Jenkinsfile (hunk 1 above)
Comment (akoushke) [reply Jenkinsfile]: will need to talk more about this tomorrow.

**Comment by akoushke** [comment]
> Might be helpful to have these folder checks as npm scripts as well so we can manually run them easier.

yeah that should not be a problem

**Comment by akoushke** [comment]
@adamweeks take a look please.

File: Jenkinsfile (hunk 2 above)
Comment (adamweeks) [reply Jenkinsfile]: Instead of creating a new stage, this verification step should be in the `Build for CDN` stage instead.

File: package.json [hunk 3]
```diff
@@ -27,6 +27,8 @@
     "static-analysis": "npm run eslint",
     "lint:eslint": "eslint --ignore-path .gitignore $(npm run --silent lint:eslint:ci-reporter)",
     "eslint": "eslint ./",
+    "es5-check:all": "es-check es5 packages/node_modules/@ciscospark/**/dist/*.js",
+    "es5-check:esm": "es-check es5 packages/node_modules/@ciscospark/**/esm/*.js",
```
Comment (adamweeks) [inline package.json]: This needs to be updated to reflect that esm files are being built in the `/es/` folder. 

Also, when run, it is failing:

>ES-Check Error:
          ----
          · erroring file: packages/node_modules/@ciscospark/widget-space/es/index.js
          · error: SyntaxError: The keyword 'import' is reserved (1:0)
          · see the printed err.stack below for context
          ----

Looks like it supports a `--module` flag to allow `import` statements, but it doesn't work with the `es5` flag... I'm ok with changing this to:

`es-check --module es6 packages/node_modules/@ciscospark/**/es/*.js`

File: package.json [hunk 4]
```diff
@@ -27,6 +27,8 @@
     "static-analysis": "npm run eslint",
     "lint:eslint": "eslint --ignore-path .gitignore $(npm run --silent lint:eslint:ci-reporter)",
     "eslint": "eslint ./",
+    "es5-check:all": "es-check es5 packages/node_modules/@ciscospark/**/dist/*.js",
```
Comment (adamweeks) [inline package.json]: `es5-check:dist` seems to be more consistent with what this is actually doing.

Comment (adamweeks) [inline package.json]: We have components in the `@webex` folder which should be checked as well.

File: Jenkinsfile (hunk 2 above)
Comment (akoushke) [reply Jenkinsfile]: ohhhh.. I got you. I misread the comment AGAIN.

**Comment by akoushke** [comment]
commits are not ready yet. will fix them after an approval

**Pull request merged**